src/SDKs/Azure.Base/data-plane/Azure.Base/Http/HttpPipelineOptions.cs

```diff
@@ -77,6 +79,19 @@ public void AddService(object service, Type type = null)
             }
         }
 
+        public HttpPipeline Build(Type clientType)
+        {
+            var componentAttribute = clientType.Assembly.GetCustomAttribute<AzureSdkComponentAttribute>();
+            if (componentAttribute == null)
+            {
+                throw new InvalidOperationException($"{nameof(AzureSdkComponentAttribute)} is required to be set on client SDK assembly.");
```

Comment:
Could you add more detail to the exception message: a) should state that it's the assembly of the clientType parameter, b) maybe should show the full name and path of the assembly?